test/connectivity/modules/NetworkController/FirewallOfflineController.cs

```diff
@@ -10,54 +9,37 @@ namespace NetworkController
 
     class FirewallOfflineController : IController
     {
-        static readonly ILogger Log = Logger.Factory.CreateLogger<FirewallOfflineController>();
-        readonly IFirewallCommands networkCommands;
+        static readonly ILogger Log = Logger.Factory.CreateLogger<LinuxFirewallOfflineController>();
+        readonly IController underlyingController;
 
         public FirewallOfflineController(string networkInterfaceName)
         {
-            Preconditions.CheckNonWhiteSpace(networkInterfaceName, nameof(networkInterfaceName));
-
             if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
             {
-                this.networkCommands = new WindowsFirewallCommands(networkInterfaceName);
+                this.underlyingController = new WindowsFirewallOfflineController(networkInterfaceName);
             }
             else
             {
-                this.networkCommands = new LinuxFirewallCommands(networkInterfaceName);
+                this.underlyingController = new LinuxFirewallOfflineController(networkInterfaceName);
             }
         }
 
         public string Description => "FirewallOffline";
 
-        public async Task<NetworkStatus> GetStatus(CancellationToken cs) => await this.networkCommands.GetStatus(cs);
-
-        public Task<bool> SetStatus(NetworkStatus status, CancellationToken cs)
-        {
-            switch (status)
-            {
-                case NetworkStatus.Restricted:
-                    return this.AddNetworkControllingRule(cs);
-                case NetworkStatus.Default:
-                    return this.RemoveNetworkControllingRule(cs);
-                default:
-                    Log.LogDebug($"Status not set {status}");
-                    throw new NotSupportedException($"Status is not supported {status}");
-            }
-        }
+        public Task<NetworkStatus> GetStatus(CancellationToken cs) => this.underlyingController.GetStatus(cs);
 
-        async Task<bool> AddNetworkControllingRule(CancellationToken cs)
+        public async Task<bool> SetStatus(NetworkStatus status, CancellationToken cs)
         {
-            bool result = await this.networkCommands.AddDropRule(cs);
-
-            NetworkStatus status = await this.GetStatus(cs);
-            Log.LogInformation($"Command AddDropRule execution success {result}, network status {status}");
+            bool result = await this.underlyingController.SetStatus(status, cs);
+            Log.LogInformation($"Command SetStatus {NetworkStatus.Restricted} execution success {result}, network status {status}");
```

Comment:
why print "SetStatus {NetworkStatus.Restricted}"?  I prefer not printing "success" if result is false.

Reply:
fixed

---
In reply to: [359155775](https://github.com/Azure/iotedge/pull/2137#discussion_r359155775) [](ancestors = 359155775)